repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/contract/ContractPreProcessor.java

```diff
@@ -0,0 +1,269 @@
+        entity.setAttribute(ATTR_VERSION, newVersionNumber);
+        entity.setAttribute(ATTR_ASSET_GUID, associatedAsset.getEntity().getGuid());
+
+        datasetAttributeSync(context, associatedAsset.getEntity(), contract, entity);
+
+    }
+
+    private List<String> getDiffAttributes(EntityMutationContext context, AtlasEntity entity, AtlasEntity latestExistingVersion) throws AtlasBaseException {
+        AtlasEntityComparator entityComparator = new AtlasEntityComparator(typeRegistry, entityRetriever, context.getGuidAssignments(), true, true);
+        AtlasEntityComparator.AtlasEntityDiffResult diffResult = entityComparator.getDiffResult(entity, latestExistingVersion, false);
+        List<String> attributesSet = new ArrayList<>();
+
+        if (diffResult.hasDifference()) {
+            for (Map.Entry<String, Object> entry : diffResult.getDiffEntity().getAttributes().entrySet()) {
+                if (!entry.getKey().equals(QUALIFIED_NAME)) {
+                    attributesSet.add(entry.getKey());
+                }
+            }
+        }
+        return attributesSet;
+    }
+
+    private boolean isEqualContract(String firstNode, String secondNode) throws AtlasBaseException {
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            JsonNode actualObj1 = mapper.readTree(firstNode);
+            JsonNode actualObj2 = mapper.readTree(secondNode);
+            //Ignore status field change
+            ((ObjectNode) actualObj1).remove(CONTRACT_ATTR_STATUS);
+            ((ObjectNode) actualObj2).remove(CONTRACT_ATTR_STATUS);
+
+            return actualObj1.equals(actualObj2);
+        } catch (JsonProcessingException e) {
+            throw new AtlasBaseException(JSON_ERROR, e.getMessage());
+        }
+
+    }
+
+    private void updateExistingVersion(EntityMutationContext context, AtlasEntity entity, AtlasEntity currentVersionEntity) throws AtlasBaseException {
+        removeCreatingVertex(context, entity);
+        entity.setAttribute(QUALIFIED_NAME, currentVersionEntity.getAttribute(QUALIFIED_NAME));
+        entity.setGuid(currentVersionEntity.getGuid());
+
+        AtlasVertex vertex = AtlasGraphUtilsV2.findByGuid(entity.getGuid());
+
+        AtlasEntityType entityType = ensureEntityType(entity.getTypeName());
+
+        context.addUpdated(entity.getGuid(), entity, entityType, vertex);
+
+    }
+
+    private void removeCreatingVertex(EntityMutationContext context, AtlasEntity entity) throws AtlasBaseException {
+        context.getCreatedEntities().remove(entity);
+        try {
+            RequestContext.get().setSkipAuthorizationCheck(true);
+            Set<String> guids = new HashSet<>();
+            guids.add(entity.getGuid());
+            entityStore.purgeByIds(guids);
+        } finally {
+            RequestContext.get().setSkipAuthorizationCheck(false);
+        }
+
+    }
+
+    private void contractAttributeSync(AtlasEntity entity, DataContract contract) throws AtlasBaseException {
```

Comment:
contractAttributeSync -> syncContractCertificateStatus

Reply:
Right now, it is only certificate status that we sync. The idea of this method is to sync all applicable contract attributes to metadata in future.

Reply:
But agree with you to rename this and move to a generic method in future when the usecase arrives.